sky/provision/provisioner.py

```diff
@@ -432,11 +434,13 @@ def _post_provision_setup(
             ux_utils.spinner_message(
                 'Launching - Waiting for SSH access',
                 provision_logging.config.log_path)) as status:
-
-        logger.debug(
-            f'\nWaiting for SSH to be available for {cluster_name!r} ...')
-        wait_for_ssh(cluster_info, ssh_credentials)
-        logger.debug(f'SSH Connection ready for {cluster_name!r}')
+        # If on Kubernetes, skip SSH check since the pods are guaranteed to be
+        # ready by the provisioner.
```

Comment:
```suggestion
        # If on Kubernetes, skip SSH check since the pods are guaranteed to be
        # ready by the provisioner, and we use kubectl instead of SSH to run the
        # commands and rsync on the pods. SSH will still be ready after a while
        # for the users to SSH into the pod.
```